### PR TITLE
Set up CD using GitHub Actions

### LINKED ISSUE #5: Set up continuous deployment
Currently Vocabulary is being manually pushed to `npm`. This can be automated with CircleCI so that it can automatically build and publish the package once all tests pass successfully.

Best way to do this is to use some mechanism on Github Actions to detect when a new *tag* is pushed and trigger the run the [deploy script](https://github.com/creativecommons/vocabulary/blob/master/deploy.sh). It can't be the release name. It has to be able to detect a new tag, because the tag name matches the version on package.json.

### PR DESCRIPTION
**Pull request by SaurabhAgarwala**
## Fixes

Fixes #5 by @dhruvkb 

## Description
This PR adds a new workflow for Continuous Deployment using Github Actions which will push the updated packages to npm when a new release is published.

## Technical details

Please note that the NPM Authentication Token has to be stored as a GitHub secret by the name `NPM_TOKEN` for this workflow to complete successfully

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I appreciate you doing both PRs simultaneously. I'll review the other one creativecommons/vocabulary#321 and you will need to make the corresponding changes here as well.

**Comment by SaurabhAgarwala** [comment]
@dhruvkb I have made the instructed changes in both the PRs. Please review them again.

**Comment by SaurabhAgarwala** [comment]
@dhruvkb The instructed changes have been made. Please review it again

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        node-version: '12.16.1'
+
+    - name: Install dependencies
+      run: npm install --no-audit
+
+    - name: Run build
+      run: npm run build
+
+    - name: Publish package
+      run: npm publish --access public
```
Comment (dhruvkb) [inline .github/workflows/cd.yml:27]: A similar deploy script exists in this repo as well. Please sync changes with your other PR.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (SaurabhAgarwala) [reply .github/workflows/cd.yml:27]: Yes, I will update it now as the other one has been approved. Thank You :)

**Comment by SaurabhAgarwala** [comment]
@dhruvkb The final changes have been made in both the PRs

**Review by dhruvkb — APPROVED** [review]
Waiting for the token to be added. Looks good otherwise.

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.16.1'
+
+    - name: Install dependencies
+      run: npm install --no-audit
+
+    - name: Publish package
+      run: |
+        cd $GITHUB_WORKSPACE/vocabulary/
```
Comment (dhruvkb) [inline .github/workflows/cd.yml:29]: This step is redundant. The working directory is already set to the project root.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Please open a new PR with this change in `creativecommons/vocabulary`.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (SaurabhAgarwala) [reply .github/workflows/cd.yml:29]: @dhruvkb I have a doubt. I examined the failed workflow logs, but couldn't figure out how can we say that the working directory points to the project root. Please help me.

Comment (SaurabhAgarwala) [reply .github/workflows/cd.yml:29]: @dhruvkb I have been looking for the above doubt, and now found that the working directory is the root GitHub directory by default. Hence, I have made the necessary changes and removed the redundant line in PR https://github.com/creativecommons/vocabulary/pull/338. Once a test passes with this PR merged, I will make the necessary changes in the CD workflow file of this repo as well.

**Comment by SaurabhAgarwala** [comment]
I have updated the working directory to the project root. Please review it.

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**